### DISCUSSION TITLE
Slowness issues

### DISCUSSION THREAD

**Discussion by fodelement** [OctoFarm/OctoFarm#755] [Q&A]
Hello all,

I installed OctoFarm 0.1.0 on a Pi 3 B+. After about two days or so, the system comes to a crawl and I need to reboot OctoFarm. This will fix it for a day or so, but it comes back. I only have three connected printers.  (each on their own Pi3)

![image](https://user-images.githubusercontent.com/71051725/136481887-b4132570-a83f-4d40-b3d5-88b057caad12.png)

It seems like something is taking up all the memory, This was a fresh install and I did not install anything else. I am not a Linux expert, but here is the output of PS AUX.
USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root           1  0.0  1.0 167876  9592 ?        Ss   Sep26   0:39 /sbin/init fi
root           2  0.0  0.0      0     0 ?        S    Sep26   0:03 [kthreadd]
root           3  0.0  0.0      0     0 ?        I<   Sep26   0:00 [rcu_gp]
root           4  0.0  0.0      0     0 ?        I<   Sep26   0:00 [rcu_par_gp]
root           8  0.0  0.0      0     0 ?        I<   Sep26   0:00 [mm_percpu_wq
root           9  0.0  0.0      0     0 ?        S    Sep26   1:30 [ksoftirqd/0]
root          10  0.4  0.0      0     0 ?        I    Sep26  76:51 [rcu_preempt]
root          11  0.0  0.0      0     0 ?        S    Sep26   1:29 [migration/0]
root          12  0.0  0.0      0     0 ?        S    Sep26   0:00 [idle_inject/
root          14  0.0  0.0      0     0 ?        S    Sep26   0:00 [cpuhp/0]
root          15  0.0  0.0      0     0 ?        S    Sep26   0:00 [cpuhp/1]
root          16  0.0  0.0      0     0 ?        S    Sep26   0:00 [idle_inject/
root          17  0.0  0.0      0     0 ?        S    Sep26   1:28 [migration/1]
root          18  0.0  0.0      0     0 ?        S    Sep26   2:02 [ksoftirqd/1]
root          21  0.0  0.0      0     0 ?        S    Sep26   0:00 [cpuhp/2]
root          22  0.0  0.0      0     0 ?        S    Sep26   0:00 [idle_inject/
root          23  0.0  0.0      0     0 ?        S    Sep26   1:28 [migration/2]
root          24  0.0  0.0      0     0 ?        S    Sep26   1:32 [ksoftirqd/2]
root          27  0.0  0.0      0     0 ?        S    Sep26   0:00 [cpuhp/3]
root          28  0.0  0.0      0     0 ?        S    Sep26   0:00 [idle_inject/
root          29  0.0  0.0      0     0 ?        S    Sep26   1:28 [migration/3]
root          30  0.0  0.0      0     0 ?        S    Sep26   1:31 [ksoftirqd/3]
root          33  0.0  0.0      0     0 ?        S    Sep26   0:00 [kdevtmpfs]
root          34  0.0  0.0      0     0 ?        I<   Sep26   0:00 [netns]
root          35  0.0  0.0      0     0 ?        S    Sep26   0:00 [rcu_tasks_kt
root          39  0.0  0.0      0     0 ?        S    Sep26   0:00 [kauditd]
root          40  0.0  0.0      0     0 ?        S    Sep26   0:01 [khungtaskd]
root          41  0.0  0.0      0     0 ?        S    Sep26   0:00 [oom_reaper]
root          42  0.0  0.0      0     0 ?        I<   Sep26   0:00 [writeback]
root          43  0.0  0.0      0     0 ?        S    Sep26   0:00 [kcompactd0]
root          44  0.0  0.0      0     0 ?        SN   Sep26   0:00 [ksmd]
root          91  0.0  0.0      0     0 ?        I<   Sep26   0:00 [kintegrityd]
root          92  0.0  0.0      0     0 ?        I<   Sep26   0:00 [kblockd]
root          93  0.0  0.0      0     0 ?        I<   Sep26   0:00 [blkcg_punt_b
root          94  0.0  0.0      0     0 ?        I<   Sep26   0:00 [tpm_dev_wq]
root          95  0.0  0.0      0     0 ?        I<   Sep26   0:00 [ata_sff]
root          96  0.0  0.0      0     0 ?        I<   Sep26   0:00 [md]
root          97  0.0  0.0      0     0 ?        I<   Sep26   0:00 [edac-poller]
root          98  0.0  0.0      0     0 ?        I<   Sep26   0:00 [devfreq_wq]
root          99  0.0  0.0      0     0 ?        S    Sep26   0:00 [watchdogd]
root         101  0.0  0.0      0     0 ?        S    Sep26   0:11 [kswapd0]
root         102  0.0  0.0      0     0 ?        S    Sep26   0:00 [ecryptfs-kth
root         104  0.0  0.0      0     0 ?        I<   Sep26   0:00 [kthrotld]
root         105  0.0  0.0      0     0 ?        I<   Sep26   0:00 [dwc_otg]
root         106  0.0  0.0      0     0 ?        I<   Sep26   0:00 [DWC Notifica
root         108  0.0  0.0      0     0 ?        S<   Sep26   0:00 [vchiq-slot/0
root         109  0.0  0.0      0     0 ?        S<   Sep26   0:00 [vchiq-recy/0
root         110  0.0  0.0      0     0 ?        S<   Sep26   0:00 [vchiq-sync/0
root         111  0.0  0.0      0     0 ?        I<   Sep26   0:00 [ipv6_addrcon
root         120  0.0  0.0      0     0 ?        I<   Sep26   0:00 [kstrp]
root         123  0.0  0.0      0     0 ?        I<   Sep26   0:00 [kworker/u9:0
root         128  0.0  0.0      0     0 ?        I<   Sep26   0:00 [cryptd]
root         153  0.0  0.0      0     0 ?        S    Sep26   0:00 [spi0]
root         154  0.0  0.0      0     0 ?        I<   Sep26   0:00 [charger_mana
root         165  0.0  0.0      0     0 ?        I<   Sep26   0:00 [mmc_complete
root         532  0.0  0.0      0     0 ?        S    Sep26   0:00 [irq/151-usb-
root         740  0.0  0.0      0     0 ?        I<   Sep26   0:00 [raid5wq]
root         799  0.0  0.0      0     0 ?        S    Sep26   1:24 [jbd2/mmcblk0
root         800  0.0  0.0      0     0 ?        I<   Sep26   0:00 [ext4-rsv-con
root         877  0.0  3.2 100392 29844 ?        S<s  Sep26   0:36 /lib/systemd/
root         902  0.0  0.4  20484  4144 ?        Ss   Sep26   0:06 /lib/systemd/
root         923  0.0  0.0      0     0 ?        S    Sep26   0:00 [vchiq-keep/0
root         924  0.0  0.0      0     0 ?        S<   Sep26   0:00 [SMIO]
root         951  0.0  0.0      0     0 ?        I<   Sep26   0:00 [mmal-vchiq]
root         952  0.0  0.0      0     0 ?        I<   Sep26   0:00 [mmal-vchiq]
root         953  0.0  0.0      0     0 ?        I<   Sep26   0:00 [mmal-vchiq]
root         954  0.0  0.0      0     0 ?        I<   Sep26   0:00 [mmal-vchiq]
root        1062  0.0  0.0      0     0 ?        I<   Sep26   0:00 [cfg80211]
root        1152  0.0  0.0      0     0 ?        I<   Sep26   0:00 [brcmf_wq/mmc
root        1157  0.0  0.0      0     0 ?        S    Sep26   0:00 [brcmf_wdog/m
root        1524  0.0  0.0      0     0 ?        I<   Sep26   0:00 [kaluad]
root        1525  0.0  0.0      0     0 ?        I<   Sep26   0:00 [kmpath_rdacd
root        1526  0.0  0.0      0     0 ?        I<   Sep26   0:00 [kmpathd]
root        1527  0.0  0.0      0     0 ?        I<   Sep26   0:00 [kmpath_handl
root        1528  0.0  1.7 280232 16604 ?        SLsl Sep26   2:49 /sbin/multipa
root        1537  0.0  0.0      0     0 ?        S<   Sep26   0:00 [loop0]
root        1540  0.0  0.0      0     0 ?        S<   Sep26   0:00 [loop1]
root        1545  0.0  0.0      0     0 ?        S<   Sep26   0:00 [loop2]
root        1547  0.0  0.0      0     0 ?        S<   Sep26   0:00 [loop3]
root        1549  0.0  0.0      0     0 ?        S<   Sep26   0:00 [loop4]
root        1551  0.0  0.0      0     0 ?        S<   Sep26   0:00 [loop5]
root        1552  0.0  0.0      0     0 ?        S<   Sep26   0:00 [loop6]
systemd+    1623  0.0  0.5  26476  5380 ?        Ss   Sep26   1:24 /lib/systemd/
systemd+    1625  0.0  0.8  24088  8332 ?        Ss   Sep26   0:03 /lib/systemd/
root        1658  0.0  0.6 237504  5592 ?        Ssl  Sep26   0:51 /usr/lib/acco
avahi       1659  0.0  0.3   7152  3000 ?        Ss   Sep26   2:38 avahi-daemon:
message+    1660  0.0  0.4   8236  4048 ?        Ss   Sep26   0:04 /usr/bin/dbus
root        1664  0.0  0.2  80796  2492 ?        Ssl  Sep26   2:46 /usr/sbin/irq
root        1665  0.0  1.1  29100 10532 ?        Ss   Sep26   0:00 /usr/bin/pyth
syslog      1667  0.0  0.4 220936  3792 ?        Ssl  Sep26   0:06 /usr/sbin/rsy
root        1669  0.0  2.2 1294472 21248 ?       Ssl  Sep26   3:01 /usr/lib/snap
root        1670  0.0  0.5  16512  5132 ?        Ss   Sep26   0:04 /lib/systemd/
root        1671  0.0  0.3  12376  3112 ?        Ss   Sep26   0:11 /sbin/wpa_sup
avahi       1674  0.0  0.0   6888   328 ?        S    Sep26   0:00 avahi-daemon:
root        1708  0.0  0.1  12772  1812 ?        Ss   Sep26   0:17 wpa_supplican
mongodb     1725  1.9  9.5 1589136 88628 ?       Ssl  Sep26 335:06 /usr/bin/mong
root        1731  0.0  0.2   8336  2092 ?        Ss   Sep26   0:03 /usr/sbin/cro
root        1743  0.0  1.3 107828 12608 ?        Ssl  Sep26   0:00 /usr/bin/pyth
daemon      1744  0.0  0.2   3592  1936 ?        Ss   Sep26   0:00 /usr/sbin/atd
root        1748  0.0  0.1   6836  1436 ttyS0    Ss+  Sep26   0:00 /sbin/agetty
root        1752  0.0  0.1   5312  1256 tty1     Ss+  Sep26   0:00 /sbin/agetty
root        1805  0.0  1.1  19892 10604 ?        Ss   Sep26   1:35 /usr/sbin/hap
haproxy     1806  0.0  0.7 241372  7304 ?        Sl   Sep26  13:26 /usr/sbin/hap
root        1828  0.0  0.1   2520  1640 ?        Ss   Sep26   0:44 /usr/sbin/dhc
root        1830  0.0  0.5  12208  4884 ?        Ss   Sep26   0:00 sshd: /usr/sb
root        1848  0.0  0.5 232936  5320 ?        Ssl  Sep26   0:00 /usr/lib/poli
pi          1972  0.1  4.9 894888 46056 ?        Ssl  Sep26  20:29 PM2 v5.1.0: G
pi          2010  101 16.7 987964 155408 ?       Rsl  Sep26 17487:35 node /home/
systemd+  370276  0.0  0.6  90228  6488 ?        Ssl  Oct07   0:00 /lib/systemd/
root      639324  0.1  0.0      0     0 ?        I    00:26   0:03 [kworker/2:2-
root      661723  0.0  0.0      0     0 ?        I    00:33   0:00 [kworker/u8:0
root      687701  0.0  0.0      0     0 ?        I    00:41   0:00 [kworker/u8:2
root      696321  0.1  0.0      0     0 ?        I    00:43   0:00 [kworker/3:2-
root      701942  0.0  0.0      0     0 ?        I    00:45   0:00 [kworker/1:0-
root      706167  0.0  0.0      0     0 ?        I    00:46   0:00 [kworker/0:2-
root      709308  0.0  0.0      0     0 ?        I<   00:47   0:00 [kworker/0:0H
root      714504  0.0  0.0      0     0 ?        I    00:49   0:00 [kworker/3:1-
root      719001  0.0  0.0      0     0 ?        I    00:50   0:00 [kworker/1:2-
root      722625  0.0  0.0      0     0 ?        I<   00:51   0:00 [kworker/2:2H
root      723381  0.1  0.0      0     0 ?        I    00:51   0:00 [kworker/0:0-
root      723471  0.0  0.0      0     0 ?        I<   00:51   0:00 [kworker/1:0H
root      725503  0.0  0.0      0     0 ?        I    00:52   0:00 [kworker/2:1-
root      728198  0.0  0.8  15588  7824 ?        Ss   00:53   0:00 sshd: pi [pri
pi        728410  0.3  1.0  19584 10040 ?        Ss   00:53   0:01 /lib/systemd/
pi        728412  0.0  0.6 169944  5612 ?        S    00:53   0:00 (sd-pam)
root      728414  0.0  0.0      0     0 ?        I<   00:53   0:00 [kworker/0:2H
pi        728875  0.0  0.5  15724  4928 ?        R    00:53   0:00 sshd: pi@pts/
pi        728882  0.2  0.5   9664  4676 pts/0    Ss   00:53   0:00 -bash
root      730843  0.0  0.0      0     0 ?        I<   00:53   0:00 [kworker/3:2H
root      732389  0.0  0.0      0     0 ?        I    00:54   0:00 [kworker/3:0-
root      733872  0.0  0.0      0     0 ?        I    00:54   0:00 [kworker/u8:1
root      736372  0.1  0.0      0     0 ?        I    00:55   0:00 [kworker/1:1-
root      741135  0.0  0.0      0     0 ?        I    00:57   0:00 [kworker/0:1-
root      741236  0.0  0.0      0     0 ?        I<   00:57   0:00 [kworker/1:2H
root      742883  0.0  0.0      0     0 ?        I    00:57   0:00 [kworker/2:0-
root      743315  0.0  0.0      0     0 ?        I<   00:57   0:00 [kworker/2:1H
root      746229  0.1  0.0      0     0 ?        I    00:58   0:00 [kworker/1:3-
pi        749603  0.0  0.0      0     0 ?        Z    00:59   0:00 [sh] <defunct
pi        749607  0.0  0.0      0     0 ?        Z    00:59   0:00 [sh] <defunct
pi        749609  0.0  0.0      0     0 ?        Z    00:59   0:00 [sh] <defunct
pi        749612  0.0  0.0      0     0 ?        Z    00:59   0:00 [sh] <defunct
pi        749614  0.0  0.2  10184  2788 pts/0    R+   00:59   0:00 ps aux
root      953631  0.0  0.0      0     0 ?        I<   Oct07   0:04 [kworker/2:0H
root     2764386  0.0  0.0      0     0 ?        I<   Oct07   0:02 [kworker/1:1H
root     3199240  0.0  0.0      0     0 ?        I<   Oct07   0:01 [kworker/3:0H

 I have the logs from farmpi running, but it's been spinning for over an hour. (Which I assume is from t he same issue) 

If there is anything else I can provide to help, please let me know.

Thank you!

**Comment by NotExpectedYet** [comment]
Hello, 

Do you mean the yellow part in your memory? On linux that's pretty normal, it will always use all available RAM. This doesn't look like a system issue as OctoFarm is reporting pretty standard usage there for a Pi3B+ as far as I'm aware. Something in the last update is slowing something down I believe just not sure what at the moment. 

#751 

We're tracking an issue with that. I don't think there's going to be much in the logs unless you've got any errors been produced but please feel free to upload them to that issue and I can take a look.
